nova/virt/vmwareapi/vmops.py

```diff
@@ -1552,6 +1558,49 @@ def _clean_up_after_special_spawning(self, context, instance_memory_mb,
                 placement_client.set_inventory_for_provider(context, rp.uuid,
                                                             inv_data)
 
+    def _stack_vm_to_host_if_needed(self, instance):
+        """Selects the fullest host that can fit the VM.
+
+        This runs only if DRS is disabled and the configuration
+        CONF.vmware.drs_disabled_stack_vms is set to True
+
+        :returns: HostSystem mo-ref or None
+        """
+        if not CONF.vmware.drs_disabled_stack_vms:
+            return None
+        if cluster_util.is_drs_enabled(self._session, self._cluster):
+            return None
+        LOG.info("DRS is disabled and stacking VMs was enabled.")
+
+        hosts_free = []
+        hosts_full = []
+        for host_ref, host in vm_util.get_stats_from_cluster_per_host(
+                self._session, self._cluster).items():
+            memory_mb_free = (host['memory_mb'] -
+                              host['memory_mb_used'] -
+                              host['memory_mb_reserved'])
+            if memory_mb_free < instance.memory_mb:
+                hosts_full.append((host['name'], memory_mb_free))
+                continue
+            else:
+                hosts_free.append({'host_ref': host_ref,
+                                   'name': host['name'],
+                                   'memory_mb_free': memory_mb_free})
+
+        if hosts_full:
+            LOG.debug("Placing VM with %(memory_mb)s MiB ignored hosts "
+                      "without enough free memory %(hosts_full)s",
+                      {'memory_mb': instance.memory_mb,
+                       'hosts_full': hosts_full})
+        if not hosts_free:
+            LOG.debug("Didn't find a host with enough memory to fit the VM.")
+            return None
```

Comment:
We just return `None` - maybe we should raise an exception instead, since we're tasked with scheduling the VM by setting and DRS is also disabled and thus won't put the VM anywhere sensible and we already know there's no space. Imho there's no need to continue building the VM and thus wasting resources on nova-compute and vCenter side.